web/public/js/controllers/dojo-event-form-controller.js

```diff
@@ -260,21 +261,22 @@
       }
     };
 
-    $scope.prefillDojoAddress = function() {
-      if($scope.eventInfo.prefillAddress) {
+    $scope.prefillDojoAddress = function (done) {
+      if ($scope.eventInfo.useDojoAddress) {
+        $scope.originalEvent.city = $scope.eventInfo.city;
+        $scope.originalEvent.address = $scope.eventInfo.address;
+
         $scope.eventInfo.city = $scope.eventInfo.dojoCity;
         $scope.eventInfo.address = $scope.eventInfo.dojoAddress;
 
         $scope.updateLocalStorage('city', $scope.eventInfo.dojoCity);
         $scope.updateLocalStorage('address', $scope.eventInfo.dojoAddress);
-        $scope.updateLocalStorage('prefillAddress', true);
-        return;
+        $scope.updateLocalStorage('useDojoAddress', true);
+      } else if ($scope.originalEvent.city && $scope.originalEvent.address) { // we restore to previous data
+        $scope.eventInfo.city = $scope.originalEvent.city;
+        $scope.eventInfo.address = $scope.originalEvent.address;
       }
-
-      $scope.updateLocalStorage('prefillAddress', false);
-      $scope.eventInfo.city = $scope.eventInfo.address = null;
-      $scope.updateLocalStorage('city', null);
-      $scope.updateLocalStorage('address', null);
+      if (done) return done();
```

Comment:
whats the effect of if done here?

Reply:
Usage for both waterfall and front-end.
That's why in the waterfall it's the only function using $scope